docs/diagrams/catchup-procedure.puml

```diff
@@ -0,0 +1,317 @@
+            end note
+
+            Node4 -> Node4: Discard received CONSISTENCY_PROOFs
+
+            Node4 ->> Node1: MESSAGE_REQUEST(CONSISTENCY_PROOF)
+            Node4 ->> Node2: MESSAGE_REQUEST(CONSISTENCY_PROOF)
+            Node4 ->> Node3: MESSAGE_REQUEST(CONSISTENCY_PROOF)
+            note right Node4
+                Ask other nodes for  <b>CONSISTENCY_PROOFs from
+                <b>own ledger size to medium other's ledger size
+            end note
+
+            Node1 ->> Node4: MESSAGE_RESPONSE(CONSISTENCY_PROOF)
+            Node2 ->> Node4: MESSAGE_RESPONSE(CONSISTENCY_PROOF)
+            Node3 ->> Node4: MESSAGE_RESPONSE(CONSISTENCY_PROOF)
+            note right Node4
+                Gather at least f+1  <b>same CONSISTENCY_PROOFs</b>
+            end note
+
+        end opt
+
+        note over Node1, Node4
+            Node gathered at least f+1 same CONSISTENCY_PROOFs
+        end note
+
+        Node4 -> Node4: Process stashed ORDEREDs
+        note right Node4
+            <b>Process accumulated stashed ORDEREDs:
+            For each stashed ORDERED:
+                - re-apply 3PC batch,
+                - execute 3PC batch
+        end note
+
+        alt Ledger is pool
+            Node4 -> Node4: mode = discovering
+        else Ledger is not pool
+            Node4 -> Node4: mode = syncing
+        end alt
+
+        Node4 -> Node4: Designate target ledger size for catch-up
+        note right Node4
+            Designate  <b>target ledger size</b> for catch-up
+            according to  <b>quorumed CONSISTENCY_PROOF</b>
+        end note
+
+        Node4 -> Node4: Generate CATCHUP_REQs
+        note right Node4
+            Generate <b>CATCHUP_REQs</b>  for different slices of lacking part
+            of ledger in quantity corresponding to count of other nodes
+        end note
+
+        Node4 ->> Node1: CATCHUP_REQ
+        Node4 ->> Node2: CATCHUP_REQ
+        Node4 ->> Node3: CATCHUP_REQ
+        note right Node4
+            Send generated CATCHUP_REQs to other nodes -
+            each CATCHUP_REQ is sent to one of the nodes only.
+            <b>Each CATCHUP_REQ contains:
+                - <b>seq_no_start</b> <i>(seq_no of first requested transaction),
+                - <b>seq_no_end</b> <i>(seq_no of last requested transaction),
+                - <b>catchup_till</b> <i>(seq_no of last transaction to catch up -
+                  <i>i.e. target ledger size)
+        end note
+
+        Node1 ->> Node4: CATCHUP_REP
+        Node2 ->> Node4: CATCHUP_REP
+        Node3 ->> Node4: CATCHUP_REP
+        note right Node4
+            Other nodes reply with CATCHUP_REPs.
+            <b>Each CATCHUP_REP contains:
+                - <b>transactions</b> <i>(requested transactions),
+                - <b>consistency proof</b> <i>(consistency proof from last transaction
+                  <i>in this reply to last transaction to catch up)
+        end note
+
+        loop On each received catch-up reply
```

Comment:
Should we mention how we re-request transactions from other nodes in case we didn't get a reply from a node for the given timestamp?